wordpress-mobile/WordPress-iOS
# Aztec Modal Announcement UI

This PR adds the Aztec announcement UI to WPiOS.

![aztec-announce](https://user-images.githubusercontent.com/4780/27557739-852c632c-5ab2-11e7-8c53-7cc3bc1a72b3.gif)

I've tried to make the alert relatively reusable, so that we can use it for upcoming login alerts. We'll need to make some tweaks there but I think it should be relatively straightforward. Essentially, the UI is configured by passing in a "configuration" object, which contains all of the label and button values / handlers. The view controller will update all its labels and resize as necessary.

A custom presentation is used to add a translucent view behind the modal.

To Test:

* Build and run the app (test with a device / sim with an existing build, and with a fresh install). When you first access a blog detail view controller, the alert should be displayed. Ensure it doesn't re-appear!
* Try dismissing the first modal screen.
* Try accepting the first modal screen and dismissing the second. The Aztec editor should be presented. Check also that App Settings shows that Aztec is enabled.

Notes:

* The blue links in the modals currently don't do anything, as we don't yet have an announcement post to link to.
* In a vertically compact layout (basically just 6+ / 7+ in landscape), the modal will hide its image view. This works okay with the modal content we have now, but ultimately we should probably put the whole thing in a scroll view in case it gets too long.

Needs review: @nheagy 
cc @iamthomasbishop
